.github/workflows/cont_integration.yml

```diff
@@ -46,3 +46,35 @@ jobs:
       - run: cargo check --verbose --no-default-features --features=proxy,use-openssl
       - run: cargo check --verbose --no-default-features --features=proxy,use-rustls
       - run: cargo check --verbose --no-default-features --features=proxy,use-rustls-ring
+
+  fmt:
+    name: Rust fmt
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          profile: minimal
+          components: rustfmt
+      - name: Check fmt
+        run: cargo fmt --all -- --config format_code_in_doc_comments=true --check
+
+  clippy_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: 1.78.0
+          components: clippy
+          override: true
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2.2.1
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-features --all-targets -- -D warnings
```

Comment:
Since we are touching this, can we:

1. bump all the `actions/checkout` from `v2` to `v4` in the file?
2. change all the `actions-rs/toolchain` (abandoned) to [`dtolnay/rust-toolchain`](https://github.com/dtolnay/rust-toolchain)?

Here's an [example of how `rust-bitcoin` does that in CI](https://github.com/rust-bitcoin/rust-bitcoin/blob/80671bbea8033ca3102cdc45ac5622c538173cca/.github/workflows/rust.yml#L88-L91):

```yml
      - name: "Select toolchain"
        uses: dtolnay/rust-toolchain@stable
        with:
          toolchain: "1.63.0"
```